src/cmd/initContainer.go

```diff
@@ -397,22 +398,81 @@ func mountBind(containerPath, source, flags string) error {
 	return nil
 }
 
+// redirectPath serves for creating symlinks for mainly crucial system files to
+// their counterparts on the host's filesystem.
+//
+// containerPath and target must be absolute paths
+//
+// folder signifies the target is a folder
+//
+// The files of the host's filesystem are available in a toolbox containers
+// under /run/host.
+//
+// If the target is a symlink, redirectPath will follow the chain of links and
+// try to lookup the real target in /run/host
+//
+// Example: systemd-resolved makes /etc/resolv.conf a link to
+// /run/systemd/resolved/resolv.conf - in a container found under
+// /run/host/run/systemd/resolved/resolv.conf).
 func redirectPath(containerPath, target string, folder bool) error {
 	logrus.Debugf("Redirecting %s to %s", containerPath, target)
 
-	err := os.Remove(containerPath)
+	targetInfo, err := os.Lstat(target)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("target %s was not found in the container", target)
+		}
+		return fmt.Errorf("failed to lstat %s: %w", target, err)
+	}
+	isSymlink := (targetInfo.Mode() & os.ModeSymlink) != 0
+
+	// If the target is a link, follow the chain. If needed, repeat the process.
+	for isSymlink {
+		logrus.Debugf("Path %s is a symlink. Resolving.", target)
+		newTarget, err := os.Readlink(target)
```

Comment:
Am I wrong in thinking that a lot of this can be simplified by using [filepath.EvalSymlinks](https://golang.org/pkg/path/filepath/#EvalSymlinks) ?

For example, I have:
```
$ ls -l /etc/os-release 
lrwxrwxrwx. 1 root root 21 Apr 28 14:50 /etc/os-release -> ../usr/lib/os-release
```

Invoking `filepath.EvalSymlinks("/etc/os-release")` gives me `/usr/lib/os-release`.

For the sake of sanity, I don't mind requiring that the `redirectPath` function accepts only absolute paths, because it's an internal function whose callers are under our control. If we go down this path, then I would check the inputs early on in the function and `panic` out (because it would be a programmer error) if the conditions aren't met.

Reply:
Ignoring the exact Go APIs that are used, I think that the `redirectPath` function should roughly do the following:

* If `target` is not a symbolic link, then don't bother checking if it exists or not. Just link `containerPath` to `target`.

* If `target` is a symbolic link, then check if it's resolving. If it's resolving, then link `containerPath` to `target`.

* If `target` is a symbolic link and if it's not resolving, then either the link is absolute or relative. If it's absolute, then prepend `/run/host` to the destination of `target` and link `containerPath` to it, which may or not lead to a valid file. If it's relative, then link `containerPath` to `target` even though we know that it doesn't resolve. In both these cases, we are not trying to guarantee that `containerPath` points to a valid file. We are merely doing the best we can to dig up the right destination, and giving up if we run out of guesses.

* It should not fail the entry-point process even if `target` doesn't exist or resolve. We can log a debug message or such, but we shouldn't stop the container from starting up just because there's something wrong with a configuration file. In this case, a slightly broken container that starts is vastly better than one doesn't start at all. If nothing else, it makes it easier to debug.

At the end of the day, all sane operating systems should be using relative symbolic links on the host. Fedora does that, and this is exactly the reason why relative symlinks are nice. They work even if you change the root from `/` to `/run/host`.

What we are trying to fix, are those few cases where a human created an absolute symbolic link. We should do our best to handle the obvious and simple cases, but if a human has made a mistake, then it's their responsibility to fix that up.